scheduler-driver/src/main/java/gov/nasa/jpl/aerie/scheduler/simulation/ResumableSimulationDriver.java

```diff
@@ -328,93 +338,67 @@ private void simulateSchedule(final Map<ActivityDirectiveId, ActivityDirective>
    * @return its duration if the activity has been simulated and has finished simulating, an IllegalArgumentException otherwise
    */
   public Optional<Duration> getActivityDuration(ActivityDirectiveId activityDirectiveId){
+    //potential cause of non presence: activity is outside plan bounds
```

Comment:
Another potential cause of absence is if it hasn't yet been scheduled. Is this potential cause more expected than if the activity is outside plan bounds?

Reply:
not really, you're right, I was trying to write what we learned in this case